### PR TITLE
Implement API for getting album collection status

### PR DESCRIPTION
**Pull request by ycanardeau**
#724

### CODE REVIEW

File: VocaDbModel/DataContracts/Users/AlbumForUserForApiContract.cs [hunk 1]
```diff
@@ -22,18 +22,18 @@ public class AlbumForUserForApiContract
...
-				PurchaseStatus = albumForUser.PurchaseStatus;
+				MediaType = albumForUser?.MediaType ?? null;
+				PurchaseStatus = albumForUser?.PurchaseStatus ?? null;
 			}
 
 			if (includeUser)
 			{
-				User = new UserForApiContract(albumForUser.User);
+				User = albumForUser != null ? new UserForApiContract(albumForUser.User) : null;
 			}
```
Comment (ycanardeau) [inline VocaDbModel/DataContracts/Users/AlbumForUserForApiContract.cs:37]: Should we use `EmitDefaultValue = false` for these nullable properties?

File: VocaDbModel/Database/Queries/UserQueries.cs [hunk 2]
```diff
@@ -1828,6 +1828,12 @@ public bool VerifyEmail(Guid requestId)
 			var artistForUser = ctx.OfType<ArtistForUser>().Query().FirstOrDefault(s => s.Artist.Id == artistId && s.User.Id == userId);
 			return new ArtistForUserForApiContract(artistForUser, LanguagePreference, _entryImagePersister, ArtistOptionalFields.None);
 		});
+
+		public AlbumForUserForApiContract GetAlbumForUser(int userId, int albumId) => HandleQuery(ctx =>
+		{
+			var albumForUser = ctx.OfType<AlbumForUser>().Query().FirstOrDefault(s => s.Album.Id == albumId && s.User.Id == userId);
+			return new AlbumForUserForApiContract(albumForUser, LanguagePreference, _entryImagePersister, AlbumOptionalFields.None, shouldShowCollectionStatus: true);
```
Comment (ycanardeau) [inline VocaDbModel/Database/Queries/UserQueries.cs:1835]: Let's use `AlbumOptionalFields.None` for now.

File: VocaDbWeb/Controllers/Api/UserApiController.cs [hunk 3]
```diff
@@ -822,5 +822,13 @@ public class PostStatusLimitedModel
...
 		[AuthenticatedCorsApi(System.Web.Mvc.HttpVerbs.Get)]
 		public ArtistForUserForApiContract GetArtistForUser(int artistId) => GetArtistForUser(_permissionContext.LoggedUserId, artistId);
+
+		[Route("{id:int}/albumCollectionStatuses/{albumId:int}")]
+		public AlbumForUserForApiContract GetAlbumForUser(int id, int albumId) => _queries.GetAlbumForUser(id, albumId);
+
+		[Route("current/albumCollectionStatuses/{albumId:int}")]
+		[Authorize]
+		[AuthenticatedCorsApi(System.Web.Mvc.HttpVerbs.Get)]
+		public AlbumForUserForApiContract GetAlbumForUser(int albumId) => GetAlbumForUser(_permissionContext.LoggedUserId, albumId);
```
Comment (ycanardeau) [inline VocaDbWeb/Controllers/Api/UserApiController.cs:832]: What should be the URL again?

File: VocaDbModel/Database/Queries/UserQueries.cs (hunk 2 above)
Comment (riipah) [reply VocaDbModel/Database/Queries/UserQueries.cs:1835]: Less is better at first

File: VocaDbWeb/Controllers/Api/UserApiController.cs (hunk 3 above)
Comment (ycanardeau) [reply VocaDbWeb/Controllers/Api/UserApiController.cs]: I decided to use kebab-case for newer APIs. `/api/users/{id:int}/followedArtists/{artistId:int}` and `/api/users/current/followedArtists/{artistId:int}` (#725) are camelCase for backward compatibility reasons with other URLs.

Comment (ycanardeau) [reply VocaDbWeb/Controllers/Api/UserApiController.cs]: I decided to use kebab-case for newer APIs. `/api/users/{id:int}/followedArtists/{artistId:int}` and `/api/users/current/followedArtists/{artistId:int}` (#725) are camelCase for backward compatibility reasons with other URLs.

**Pull request merged**